### ISSUE TITLE
Multi-byte named capture groups do not parse

### ISSUE THREAD

**Issue by dgollahon** [ammar/regexp_parser#76]
Hi again,

I have found one interesting case where `regexp_parser` will not parse one that MRI accepts:

```ruby
/(?<æ>.)(.)(?<b>\d+)(\d)/.match('ab12').named_captures # => {"æ"=>"a", "b"=>"1"}

Regexp::Parser.parse(/(?<æ>.)(.)(?<b>\d+)(\d)/) # => Regexp::Scanner::InvalidGroupOption: Invalid group option  in (?
```

See [this line](https://github.com/ruby/spec/blob/master/core/matchdata/begin_spec.rb#L100) and [this line](https://github.com/ruby/spec/blob/master/core/matchdata/end_spec.rb#L100) from `ruby/spec` for where I sourced this case.

Like in #75 I'm just curious if this is intended behavior and, if so, if it can be documented. Otherwise parity with MRI is preferred.

Thanks again!

**Comment by jaynetics** [comment]
@dgollahon This was definitely a bug.

I've released v2.0.1 with a fix.

Thank you for the excellent quality checkup!

**Comment by dgollahon** [comment]
Once again, thanks for the prompt fix!
